webshop-backend/src/shopping/ShoppingCart.java

```diff
@@ -0,0 +1,30 @@
+package shopping;
+
+import java.util.List;
+
+public class ShoppingCart {
+    private List<Product> products;
+    private int totalPrice;
+    private User owner;
+
+    public ShoppingCart(List<Product> products, int totalPrice, User owner) {
+        this.products = products;
+        this.totalPrice = totalPrice;
+        this.owner = owner;
+    }
+
+    public void addProduct(Product p) {
+        if (p != null) {
+            products.add(p);
+        }
+    }
+
+    public void removeProduct(Product p) {
+        int pieces = p.getStock();
```

Comment:
You are checking the NPE in the previous method, in the `addProduct`, but not validating here. Are we expecting products to be `null` or not?